src/shared.js

```diff
@@ -795,8 +795,18 @@ function isNewline(node) {
   return node.type === "Newline";
 }
 
-function getSource(node) {
-  const source = node.source.value;
+function getSource(sourceCode, node) {
+  let source;
+  switch (node.type) {
+    case "TSImportEqualsDeclaration": {
+      source = `= ${sourceCode.text.slice(...node.moduleReference.range)}`;
```

Comment:
Ah, you turn `Foo.Bar.Baz` into the string `"Foo.Bar.Baz"` a bit as if you had written `from "Boo.Bar.Baz"`, so that it fits into the existing sorting code. Clever!

Here are some edge cases:

- Comments: `import A = A/**/.B.C;`. I guess nobody writes code like that, but this is the level of quality I maintain in the rest of the plugin.
- `import a = require("a")` – that will also be turned into a string, and I think it’s a bit weird.

I need to think about this. In the meantime, I recommend forking, or making a separate rule that moves this problematic import-like statements out of the way.

Reply:
what is the expected behavior in these edge cases?

Reply:
It seems I have covered these edge cases